docs/v3/core/index.rst

```diff
@@ -440,47 +460,63 @@ Array metadata
 --------------
 
 Each Zarr array in a hierarchy must have an array metadata document, named
-``zarr.json``. This document must contain a single object with the following
+``zarr.json``.
+
+Mandatory
+^^^^^^^^^
+
+This document must contain a single object with the following
 mandatory names:
 
+.. _array-metadata-zarr-format:
+
 ``zarr_format``
-^^^^^^^^^^^^^^^
+""""""""""""""""
 
     An integer defining the version of the storage specification to which the
     array store adheres, must be ``3`` here.
 
+.. _array-metadata-node-type:
+
 ``node_type``
-^^^^^^^^^^^^^^^
+"""""""""""""""
 
     A string defining the type of hierarchy node element, must be ``array``
     here.
 
+.. _array-metadata-shape:
+
 ``shape``
-^^^^^^^^^
+"""""""""
 
     An array of integers providing the length of each dimension of the
     Zarr array. For example, a value ``[10, 20]`` indicates a
     two-dimensional Zarr array, where the first dimension has length
     10 and the second dimension has length 20.
 
+.. _array-metadata-data-type:
+
 ``data_type``
-^^^^^^^^^^^^^
+"""""""""""""
 
     The data type of the Zarr array. If the data type is defined in
     this specification, then the value must be the data type
     identifier provided as a string. For example, ``"float64"`` for
     little-endian 64-bit floating point number.
 
-    The ``data_type`` value is an extension point and may be defined by a data
+    The ``data_type`` value is an :ref:`extension point<extensions_section>`
+    and may be defined by a data
     type extension. If the data type is defined by an extension, then the value
-    may be either a plain string or an object containing the members ``name``
+    may be either a plain string (incl. URI) or an object containing the members ``name``
```

Comment:
```suggestion
    may be either a string or an object containing the members ``name``
```
a URI _is_ a string, so I don't think we need to mention URIs here without a compelling reason.

Reply:
And if we give this particular type a string a name ("extension identifier"?) since it's defined as:

`(^[a-z0-9-_.]+$|^https?:\/\/[^/?#]+[^?#]*$)`

Reply:
ah i didn't see that this is already in the context of data type extensions. in that case, we should be able to say that it must be a data type identifier, perhaps linking to the definition of that type.